File: framework/data-jpa/src/main/java/cn/bestwu/simpleframework/data/jpa/BaseRepository.java
package cn.bestwu.simpleframework.data.jpa;

import cn.bestwu.simpleframework.data.jpa.query.RecycleQuerydslPredicateExecutor;
import org.springframework.data.querydsl.QuerydslPredicateExecutor;
import org.springframework.data.repository.NoRepositoryBean;

/**
 * @param <T> T
 * @param <ID> ID
 * @author Peter Wu
 */
@NoRepositoryBean
public interface BaseRepository<T, ID> extends JpaExtRepository<T, ID>,
    QuerydslPredicateExecutor<T>, RecycleQuerydslPredicateExecutor<T> {

}

File: data-jpa/src/main/kotlin/top/bettercode/summer/data/jpa/BaseRepository.kt
package top.bettercode.summer.data.jpa

import org.springframework.data.repository.NoRepositoryBean

/**
 * @param <T>  T
 * @param <ID> ID
 * @author Peter Wu
</ID></T> */
@NoRepositoryBean
interface BaseRepository<T, ID> : JpaExtRepository<T, ID> 